app/components/list_filter_component.html.erb

```diff
@@ -0,0 +1,150 @@
+<div
+  data-controller="list-filter"
+  data-list-filter-selection-outlet="#samples-table"
+>
+  <%= viral_dialog(closable: false) do |dialog| %>
```

Comment:
Any specific reason for setting the dialog to be non closable? A user can accidentally click the button and it launches the dialog but there's no way for the user to close it. You can close it using the `Apply Filter` button with an empty text field but a user would have no way of knowing that. I think we should add a `X` to close the dialog or a `Cancel` button

Reply:
This was decided by @ericenns and myself when we did the project samples table. The  reason for this is that the `tags` are actually the form  `inputs` if a user removes one and then closes the dialog without the table update then the next time the table changes (sort, paging, etc...) the list to filter by will be different, loading the samples into the table.

Reply:
That makes sense but I still think we need to revisit this and see if we can solve it another way. To me it's just a bad user experience to have to apply the filter when they might have just accidentally launched the modal. I'll approve for now and we can revisit whenever we have a refactor phase